### PR TITLE
#1: Build and push all dependencies image to Dockerhub

### LINKED ISSUE #1: Experimental Image
Create a docker image with a minimal set of dependencies -  in order to build NimbleSM in serial.

### PR DESCRIPTION
**Pull request by pierrepebay**
Fixes #1

### CODE REVIEW

**Review by stmcgovern — APPROVED** [review]
This looks good as is. Some future changes will be required in light of the CI changes pursued in NimbleSM PR346.

**Review by nmm0 — COMMENTED** [review]
Looking good -- do we need the extra github actions (the commit format and so on)?

**Comment by pierrepebay** [comment]
> Looking good -- do we need the extra github actions (the commit format and so on)?

They're used on other projects so that's why I brought them here, but I'll let you decide if we keep them or not.

**Review by nmm0 — CHANGES_REQUESTED** [review]
Thanks for doing this -- overall I think we should change two things:

1. Let's get rid of the extra checks like the signed commit checking and so on since we don't need those
2. Can you build, but not push, the docker image on PRs? Right now it looks like the workflow only runs on main and always pushes. This way with PRs we can verify that the dependencies are working

**Review by nmm0 — CHANGES_REQUESTED** [review]
Woops one thing I missed, is the different environments have the same prefix for view. I think that might cause some issues with the different environments conflicting

**Review by nmm0 — CHANGES_REQUESTED** [review]
Scratch previous comment, lets just disable views since we are not using

```yaml
view: False
```

**Review by nmm0 — APPROVED** [review]
Great work, thank you!

**Comment by nmm0** [comment]
I will switch this over to my Sandia Dockerhub account before merging just FYI

File: .github/workflows/build-and-push-docker-image.yml
Comment (nmm0) [inline .github/workflows/build-and-push-docker-image.yml]:
```suggestion
          images: nmm0/nimblesm-base-env
```

**Pull request merged**
